allensdk/brain_observatory/demixer.py

```diff
@@ -120,19 +121,25 @@ def demix_time_dep_masks(raw_traces: np.ndarray, stack: np.ndarray,
         (t, n), where `t` is the number of time points and `n` is the
         number of masks.
     :param stack: 3d array representing a 1p recording movie, of
-        dimensions (t, H, W).
+        dimensions (t, H, W) or corresponding hdf5 dataset.
     :param masks: 3d array of binary roi masks, of shape (n, H, W),
         where `n` is the number of masks, and HW are the dimensions of
         an individual frame in the movie `stack`.
+    :block_size: int representing maximum number of movie frames to read
+        at a time (-1 for full length `t` of `stack`) (the default is 1000)
     :return: Tuple of demixed traces and whether each frame was skipped
         in the demixing calculation.
     """
     N, T = raw_traces.shape
     _, x, y = masks.shape
     P = x * y
 
-    if len(stack.shape) == 3:
-        stack = stack.reshape(T, P)
+    if block_size == -1:
+        block_size = T
+    elif block_size < 1:
```

Comment:
also add exception checking for case when block_size is greater than T, ie: greater than the trace length
```python
elif block_size < 1 or block_size > T:

```

Reply:
The code currently handles this case without errors, however I realize it does not do so transparently.

I would propose not to throw an exception, for the following reason: if we do not want to constrain `block_size` to be a factor of `T` (I think this would be unnecessarily constraining), then the final block will almost always be shorter than `block_size`. The case where `block_size > T` is a special case of this, where the final block is the only block, and it is shorter than `block_size`.

Instead, I would propose that I implement these 3 changes to make this clearer.
1) **Rename `block_size` to `max_block_size`**: In the docstring, `block_size` is described as the **maximum** number of movie frames to read at a time. The modified variable name would be more transparent.
2) **Make stack indexing clearer**: [L156](https://github.com/colleenjg/AllenSDK/blob/GH-1461/feature/demixer-stack-block/allensdk/brain_observatory/demixer.py#L156) calculates the current block size, `block_T` (which is then used when reshaping the block on the next line). Currently, [L157](https://github.com/colleenjg/AllenSDK/blob/GH-1461/feature/demixer-stack-block/allensdk/brain_observatory/demixer.py#L157) loads the block using `stack[t : t+block_size]`. This indexing does work even when `t+block_size > T`, but it would be more transparent/cleaner to use `t+block_T` instead. 
**L157 change:**
\- `stack_block = stack[t : t+block_size].reshape(block_T, P)`
\+ `stack_block = stack[t : t+block_T].reshape(block_T, P)`
3) **Add test case**: Add a pytest case where `block_size > T`.

Reply:
I agree with all these changes, thanks for putting thought into this! Once that's implemented I'll approve this.

Reply:
Thank you! I will push an updated commit ASAP.

Reply:
The changes have been made, and I squashed the commits.